components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/test/common/testng/CarbonBasedTestListener.java

```diff
@@ -155,6 +158,12 @@ public void onBeforeClass(ITestClass iTestClass, IMethodInstance iMethodInstance
             MockInitialContextFactory
                     .initializeDatasource(withH2Database.jndiName(), realClass, withH2Database.files());
         }
+        if (annotationPresent(realClass, WithOAuthServerConfiguration.class)) {
```

Comment:
Do we need this? When we use WithCarbonHome you should be able to use it.

Reply:
I have added this config to the identity.xml and updated the PR.